src/api/init-api.js

```diff
@@ -0,0 +1,17 @@
+const express = require('express');
+const app = express();
+const auth = require('./routes/auth');
+const cors = require('cors');
+
+function initApi(c) {
+  app.use(cors());
+  app.listen(c.prefsManager.server_port, function () {
+    console.log('Api-Server is running on port: ' + c.prefsManager.server_port);
+  });
+
```

Comment:
Can we assign the context object ("c") to app.locals here?

In this case you will not need to initialize prefsManager every time in src/api/controllers/auth.js file, and instead just access it through req.app.locals.c.prefsManager. By the way, I would suggest to rename "c" to "context" here and in app.locals, for better readability.

This also should become helpful in the future, when you call other context objects from the routes (like commands-parser.js).

For further reading on the topic:
https://expressjs.com/ru/api.html
https://stackoverflow.com/questions/28748337/whats-the-best-way-to-access-the-express-app-object-from-inside-a-separate-ro